### ISSUE TITLE
`scan_data()` is very slow on large datasets

### ISSUE THREAD

**Issue by hadley** [rstudio/pointblank#550]
Specifically the interactions and correlations sections:

```R
pointblank::scan_data(ggplot2::diamonds, sections = "IC")
```

Iteractions takes ~30s which isn't too surprising, but correlations takes a very long time, which does surprise me.

**Comment by rich-iannone** [comment]
Agreed, we could definitely make this faster/better. (Part of the entire `scan_data()` overhaul.)

**Comment by kmasiello** [comment]
Model improvements to `scan_data()` off of the python pointblank's `DataScan` to create a LLM-ingestible summary.
